src/main/java/org/broadinstitute/hellbender/tools/HaplotypeCallerSpark.java

```diff
@@ -222,13 +239,19 @@ private void writeVariants(JavaRDD<VariantContext> variants) {
      * @param reads Rdd of {@link GATKRead}
      * @return a Rdd of reads grouped into potentially overlapping shards
      */
-    private static JavaRDD<Shard<GATKRead>> createReadShards(final Broadcast<OverlapDetector<ShardBoundary>> shardBoundariesBroadcast, final JavaRDD<GATKRead> reads) {
+    private static JavaRDD<Shard<GATKRead>> createReadShards(final Broadcast<OverlapDetector<ShardBoundary>> shardBoundariesBroadcast, final JavaRDD<GATKRead> reads, final SAMFileHeader header) {
         final JavaPairRDD<ShardBoundary, GATKRead> paired = reads.flatMapToPair(read -> {
             final Collection<ShardBoundary> overlappingShards = shardBoundariesBroadcast.value().getOverlaps(read);
             return overlappingShards.stream().map(key -> new Tuple2<>(key, read)).iterator();
         });
         final JavaPairRDD<ShardBoundary, Iterable<GATKRead>> shardsWithReads = paired.groupByKey();
-        return shardsWithReads.map(shard -> new SparkReadShard(shard._1(), shard._2()));
+        return shardsWithReads.map(shard -> new SparkReadShard(shard._1(), sortReadsInMemory(shard._2(), header)));
```

Comment:
@tomwhite Why do we need to sort the reads here?  Shouldn't the reads be sorted within shards if they input file was sorted?